apps/api-documenter/src/documenters/YamlDocumenter.ts

```diff
@@ -677,11 +650,46 @@ export class YamlDocumenter {
 
   private _getYamlItemName(apiItem: ApiItem): string {
     if (apiItem.parent && apiItem.parent.kind === ApiItemKind.Namespace) {
-      // For members a namespace, show the full name excluding the package part:
-      // Example: excel.Excel.Binding --> Excel.Binding
-      return this._getUid(apiItem).replace(/^[^.]+\./, '');
+      // If the immediate parent is a namespace, then add the namespaces to the name.  For example:
+      //
+      //   // Name: "N1"
+      //   export namespace N1 {
+      //     // Name: "N1.N2"
+      //     export namespace N2 {
+      //       // Name: "N1.N2.f(x,y)"
+      //       export function f(x: string, y: string): string {
+      //         return x + y;
+      //       }
+      //
+      //
+      //       // Name: "N1.N2.C"
+      //       export class C {
+      //         // Name: "member(x,y)"  <===========
+      //         public member(x: string, y: string): string {
+      //           return x + y;
+      //         }
+      //       }
+      //     }
+      //   }
+      //
+      // In the above example, "member(x, y)" does not appear as "N1.N2.C.member(x,y)" because YamlDocumenter
```

Comment:
Any concern with TOC collisions with `class C { static x; x; }`? Both would have the TOC display name of `x` with no disambiguation.

Reply:
Does DocFX create TOC entries for class members?  I didn't think it did.

Reply:
I'm not sure at the moment, but you may have to disambiguate other items in the TOC, such as merging a function or a variable and an interface with the same name. I had to workaround this for esfx: https://github.com/esfx/esfx/blob/bbc36650efa6a88c9d998b0dafb53e8421513898/scripts/docs/yamlDocumenter.js#L214-L216